src/_DataStructures_/Trees/Trie/all-words-in-trie/all-words-in-trie.test.js

```diff
@@ -13,4 +13,49 @@ describe('Data Structure : Trie : All Words In Tree', () => {
     const expected = ['apple', 'ball', 'bed', 'bed', 'java', 'javascript'];
     expect(expected).toEqual(result);
   });
+
+  it('Should retain duplicates', () => {
+    const words = ['bed', 'bed', 'bed'];
+    const trie = new Trie();
+
+    words.forEach(word => trie.insert(word));
+
+    const result = allWordsInTrie(trie.root);
+    expect(result.length).toBe(3);
+  });
+
+  it('passing an empty array of words returns an empty array', () => {
+    const words = [];
+    const trie = new Trie();
+
+    words.forEach(word => trie.insert(word));
+
+    const result = allWordsInTrie(trie.root);
+    expect(result).toEqual([]);
+  });
+
+  it('passing an empty Trie will throw an error ', () => {
+    const trie = new Trie();
+    expect(() => {
+      allWordsInTrie(trie);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+  it('passing an empty array will throw an error ', () => {
+    expect(() => {
+      allWordsInTrie([]);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+  it('passing null will throw an error ', () => {
+    expect(() => {
+      allWordsInTrie([]);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
+
+  it('passing an array not in a Trie will throw an error ', () => {
+    expect(() => {
+      allWordsInTrie(['bed', 'ball', 'apple']);
+    }).toThrow('Cannot read property \'0\' of undefined');
+  });
```

Comment:
It's great to see you @ChristieRobson, you seem to be a quick learner. I appreciate your efforts and I will encourage you to pick more issues from the pool. 

For the above selected chunk of code, please follow the same instructions given in #149

Reply:
Tests updated with specifc error